src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -478,22 +485,76 @@ BoundExpression createArray(BoundCollectionExpression node, ArrayTypeSymbol arra
             }
         }
 
+        private BoundExpression VisitAndRewriteCollectionArgumentsSignatureOnlyMethodCall(BoundCall node, NamedTypeSymbol collectionType)
```

Comment:
Looking at `LocalRewriter.VisitObjectCreation`, I see there's special handling for expression tree scenarios and for instrumentation. Consider adding follow-up comment if we need similar handling here

Reply:
Collection expressions in general are not supported in expression trees, and we have asserts in the callers of this method. I've added an assert here as well.

For instrumentation, this is consistent with existing cases in collection expression rewriting where we do not instrument calls to well-known methods used to instantiate and populate collections since the calls are implementation details rather than user code.